.pre-commit-config.yaml

```diff
@@ -4,6 +4,10 @@ repos:
     hooks:
     -   id: double-quote-string-fixer
     -   id: end-of-file-fixer
+        exclude: |
+            (?x)^(
+                tests/engine/test_calc_job/.*|
+            )$
```

Comment:
Uhm, I see the issue. I think I would adapt the test which is more particular rather than start polluting global config files with exceptions, but I'll admit this is a somewhat personal preference. I am thinking of something like this (I took it from [here](https://stackoverflow.com/a/275025), didn't check if it actually works, just to give you an idea):

``` python
    with open(os.path.join(folder_name, submit_script_filename)) as handle:
        content = handle.read()

    content = content.rstrip() + `\n` # linter compliance
    file_regression.check(content, extension='.sh')
```

But yeah, I will agree that is also not ideal. What do you think? @sphuber do you have any opinions on this, is there a better alternative we are missing?

Reply:
In a sense yes, it would be good to adapt the test to _not_ be sensitive to trailing whitespace, because we don't really care. Maybe you can actually use `file_regression.check` itself to ignore the whitespace. See [the documentation](https://pytest-regressions.readthedocs.io/en/latest/api.html#file-regression) which shows there is a `newline` argument. Not 100% how it works but it could be useful